src/bin/segment-keyspaces.rs

```
@@ -0,0 +1,99 @@
+use anyhow::Result;
+use clap::{Parser, Subcommand};
+use segment_rs::{
+    client::Client,
+    command::{Command, CommandError},
+    connection::{Connection, ConnectionOptions},
+};
+use segment_utils::tokenizer;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(short = 'H', long)]
+    host: Option<String>,
+
+    #[arg(short, long)]
+    port: Option<u16>,
+
+    #[command(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    Export,
+    Restore { filename: String },
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = Args::parse();
+
+
+    let client = Client::new(ConnectionOptions::new(
+        &args.host.unwrap_or(String::from("127.0.0.1")),
+        args.port.unwrap_or(1698)
+    ));
+    let mut conn = client.get_connection().await?;
+
+
+    match &args.command {
+        Commands::Export => {
+            export_keyspaces(&mut conn).await;
+        }
+        Commands::Restore { filename } => {
+            restore_keyspaces(Path::new(filename), &mut conn).await;
+        }
+    }
+    Ok(())
+}
+
+async fn restore_keyspaces(file: &Path, conn: &mut Connection) {
```

let's return an error from this function instead of using `unwrap` or `expect`